src/sysl/importers/import_openapi.py

```diff
@@ -0,0 +1,520 @@
+                        # - any string matching the pattern "^x-(.*)$""
+                        # ref: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#responses-object
+
+                        includeResponses(responses)
+
+    def writeDefs(self, oaSpec, w):
+        w()
+        w('#' + '-' * 75)
+        w('# definitions')
+
+        if not oaSpec.get('components', None):
+            return
+        if not oaSpec['components'].get('schemas', None):
+            return
+
+        def _collectExternalsSoReferencesAreCorrect():
+            for (tname, tspec) in sorted(oaSpec['components']['schemas'].iteritems()):
+                properties = extract_properties(tspec)
+
+                if properties or (tspec.get('type') == 'array') or ('enum' in tspec):
+                    continue
+                externAlias['EXTERNAL_' + tname] = 'string'
+
+        _collectExternalsSoReferencesAreCorrect()
+
+        for (tname, tspec) in sorted(oaSpec['components']['schemas'].iteritems()):
+            properties = extract_properties(tspec)
+            w()
+
+            if properties:
+                w('!type {}:', tname)
+            elif tspec.get('type') == 'array' or 'enum' in tspec:
+                w('!alias {}:', tname)
+
+            requiredFields = []
+            if 'required' in tspec:
+                requiredFields = tspec['required']
+
+            with w.indent():
+                def getfields(fname, ftype, isArray=False):
+                    fieldTemplate = '{} <: {}:'
+                    ftypeSyntax = ftype
+
+                    if isArray:
+                        fieldTemplate = '{}{}'
+                    elif fname not in requiredFields:
+                        ftypeSyntax = ftype + '?'
+
+                    return fieldTemplate.format(
+                        fname if fname not in SYSL_TYPES else fname + '_',
+                        ftypeSyntax)
+
+                if properties:
+                    for (fname, fspec) in sorted(properties.iteritems()):
+                        (ftype, fdescr) = self.parse_typespec(fspec, fname, tname)
+                        w(getfields(fname, ftype))
+                        with w.indent():
+                            w(self.getTag(fname, 'json_tag'))
+                            if fdescr is not None and len(fdescr) > 0:
+                                w(self.getTag(fdescr, 'description'))
+                # handle top-level arrays
+                elif tspec.get('type') == 'array':
+                    (ftype, fdescr) = self.parse_typespec(tspec, '', tname)
+                    w(getfields('', ftype, True))
+                    with w.indent():
+                        if fdescr is not None and len(fdescr) > 0:
+                            w(self.getTag(fdescr, 'description'))
+                elif 'enum' in tspec:
+                    w(TYPE_MAP[tspec.get('type')])
+
+        index = 0
+        while len(typeSpecList) > index:
```

Comment:
```python
for typeSpec in typeSpecList:
```

Reply:
Changed and made appropriate substitutions.